src/Std/Data/DHashMap/AdditionalOperations.lean

```diff
@@ -38,6 +38,33 @@ theorem WF.map [BEq α] [Hashable α] {m : Raw α β} (h : m.WF) {f : (a : α) 
     (m.map f).WF := by
   simpa only [map_eq h] using .wf (Raw₀.map f ⟨m, h.size_buckets_pos⟩).2 (Raw₀.wfImp_map (WF.out h))
 
+/-- Internal implementation detail of the hash map -/
+@[inline] def pmapEntries {P : (a : α) → (b : β a) → Prop }
+    (f : (a : α) → (b : β a) → P a b → ((a' : α') × β' a'))
```

Comment:
Thinking about it, I don't think there is really a reason to have the predicate `P` here any more. Just take `⟨a, b⟩ ∈ m.toList` in `f`, and if a user need something else that follows from that, they can just do it in `f`.